docs/features/software-catalog/descriptor-format.md

```diff
@@ -0,0 +1,295 @@
+
+Their purpose is mainly, but not limited, to reference into external systems. This could
+for example be a reference to the git ref the entity was ingested from, to monitoring
+and logging systems, to pagerduty schedules, etc. Users may add these to descriptor
+YAML files, but in addition to this automated systems may also add annotations, either
+during ingestion into the catalog, or at a later time.
+
+Both the key and the value are strings, subject to the following restrictions.
+
+Keys have an optional prefix followed by a slash, and then the name part which is required.
+The prefix must be a valid lowercase domain name if specified, at most 253 characters in total.
+The name part must be sequences of `[a-zA-Z0-9]` separated by any of `[-_.]`, at most 63 characters
+in total.
+
+The `backstage.io/` prefix is reserved for use by Backstage core components.
+
+Values can be of any length, but are limited to being strings.
+
+## Kind: Component
+
+Describes the following entity kind:
+
+| Field        | Value                  |
+| ------------ | ---------------------- |
+| `apiVersion` | `backstage.io/v1beta1` |
+| `kind`       | `Component`            |
+
+A Component describes a software component. It is typically intimately linked to the source code
+that constitutes the component, and should be what a developer may regard a "unit of software",
+usually with a distinct deployable or linkable artifact.
+
+Descriptor files for this kind may look as follows.
+
+```yaml
+apiVersion: backstage.io/v1beta1
+kind: Component
+metadata:
+  name: artist-web
+  description: The place to be, for great artists
+spec:
+  type: website
+  lifecycle: production
+  owner: artist-relations@example.com
+```
+
+In addition to the [common envelope metadata](#common-to-all-kinds-the-metadata) shape, this kind has the
+following structure.
+
+### `apiVersion` and `kind` [required]
+
+Exactly equal to `backstage.io/v1beta1` and `Component`, respectively.
+
+### `spec.type` [required]
+
+The type of component as a string, e.g. `website`. This field is required.
+
+The software catalog accepts any type value, but an organisation should take great
+care to establish a proper taxonomy for these. Tools including Backstage itself may
+read this field and behave differently depending on its value. For example, a website
+type component may present tooling in the Backstage interface that is specific to just
+websites.
+
+The current set of well-known and common values for this field is:
+
+- `service` - a backend service, typically exposing an API
+- `website` - a website
+- `library` - a software library, such as an NPM module or a Java library
+
+### `spec.lifecycle` [required]
+
+The lifecyle state of the component, e.g. `production`. This field is required.
+
+The software catalog accepts any lifecycle value, but an organisation should take great
+care to establish a proper taxonomy for these.
+
+The current set of well-known and common values for this field is:
```

Comment:
What does this mean? It means that other values are not allowed?
FWIW, we were thinking of adding this (inside metadata) and our values also included `pre_production` for all those things that passed experimental, probably they are in some sandbox or develop environment somewhere but not in production yet.

Reply:
Sounds good!

It's surprisingly hard to nail the semantics of flags like these. They can mean different things from different perspectives/organisations, and based on one's previous experiences and preferences.

When talking about preprod, what can one read into that?

- That it's "physically" in the preprod (staging like) environment - as in, it speaks of deployment states?
- That it has reached that certain step before publicly-reachable production in the business processes of the org - as in, it speaks of ways of working?
- That it is considered not-yet-fully-stable but can be consumed if you are willing to accept the consequences - as in, it speaks of maturity or SLAs?

Or even worse - a combination of the above?

This is not me bashing on the choice of `pre_production` at all, because I am certain it makes perfect sense within your context. But I think it can be interesting to mention, and this is the reason that we need to put up a few well known values like these and be very clear about what their intended usage is, while also letting people use other values as they see fit.

Overall - thanks for your feedback :)